contributed_definitions/nyaml/NXfit.yaml

```diff
@@ -0,0 +1,345 @@
+          must have the same number of points as the corresponding dimension in
+          the ``input_independent`` field.
+        dim: []
+    residual(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        Difference between the envelope and the `input_independent` data to be fitted.
+      dimensions:
+        rank: dimRank
+        doc: |
+          The ``residual`` field must have the same rank (``dimRank``)
+          as the ``input_independent`` field. Each individual dimension of ``residual``
+          must have the same number of points as the corresponding dimension in
+          the ``input_independent`` field.
+        dim: []
+  peakPEAK(NXpeak):
+    doc: |
+      One peak of the peak model.
+      If there is no characteristic name for each peak component, is envisioned that peaks
+      are labeled as peak_0, peak_1, and so on.
+    total_area(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        Total area under the curve (can also be used for the total area minus any
+        background values).
+    relative_sensitivity_factor(NX_NUMBER):
+      unit: NX_UNITLESS
+      doc: |
+        Relative sensitivity for this peak, to be used for quantification in
+        an NXprocess.
+        
+        As an example, in X-ray spectroscopy could depend on the energy scale
+        (see position), the ionization cross section, and the element probed.
+    relative_area(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        Relative area of this peak compared to other peaks.
+        
+        The relative area can simply be derived by dividing the total_area by the
+        total area of all peaks or by a more complicated method (e.g., by
+        additionally dividing by the relative sensitivity factors). Details shall
+        be given in `global_fit_function`.
+  backgroundBACKGROUND(NXfit_background):
+    doc: |
+      One fitted background (functional form, position, and intensities) of the peak fit.
+      If there is no characteristic name for each peak component, it is envisioned that backgrounds are labeled
+      as background_0, background_1, and so on.
+  global_fit_function(NXfit_function):
+    doc: |
+      Function used to describe the overall fit to the data, taking into account the parameters of the
+      individual `NXpeak` and `NXfit_background` components.
+    formula:
+      doc: |
+        Oftentimes, if the peaks and fit backgrounds are defined independently (i.e, with their own
+        parameter sets), the resulting global fit is a function of the form
+        :math:`model = peak_1(p_1) + peak2(p_2) + backgr(p_3).`, where each :math:`p_x` describes the
+        set of parameters for one peak/background.
+  error_function(NXfit_function):
+    doc: |
+      Function used to optimize the parameters during peak fitting.
+    description:
+      doc: |
+        Description of the method used to optimize the parameters during peak fitting.
```

Comment:
extend this a bit, say if it's linear or non-linear and which algorithm was used.